get_global_param.c

```diff
@@ -102,11 +102,41 @@ void ProgramState::initGrid(const std::vector<cell_info_struct>& cells) {
         }
       }
     }
-    global_param.gridNumLatDivisions = ((largestLat - global_param.gridStartLat) / global_param.gridStepLat) + 1;
-    global_param.gridNumLonDivisions = ((largestLon - global_param.gridStartLon) / global_param.gridStepLon) + 1;
+    global_param.gridNumLatDivisions = ((global_param.gridEndLat - global_param.gridStartLat) / global_param.gridStepLat) + 1;
+    global_param.gridNumLonDivisions = ((global_param.gridEndLon - global_param.gridStartLon) / global_param.gridStepLon) + 1;
   }
 }
 
+void ProgramState::initCellMask(const std::vector<cell_info_struct>& cells) {
+
+  char            ErrStr[MAXSTRING];
+	int total_num_cells = global_param.gridNumLatDivisions * global_param.gridNumLonDivisions;
+	valid_cell_mask = new bool [total_num_cells];
+	float lat = global_param.gridStartLat;
+	float lon = global_param.gridStartLon;
+  int cellidx = 0;
+  int count = 0;
+
+	// Check all possible lat/lon pairs in the grid and check if each exists within valid_cell_coordinates set
+	while (lat <= global_param.gridEndLat){
```

Comment:
Use `for` loops here. You're iterating over a fixed number of things, and using a `for` loop will better communicate that to the maintainer of this code.